### PR TITLE
Dp24 pretextmap update

### LINKED ISSUE #5123: [Update] PretextMap should take a fai file
### Is your feature request related to a problem? Please describe

PretextMap will index the incoming fasta file, in many cases this file will already be generated and just needs to be visible to the process (it cannot be declared to the software).

In the cases of large genomes (I have a 25Gb Newt!) this will speed up the mapping process and waste less resources as we are not indexing again.

### Describe the solution you'd like

Solution is proposed here https://github.com/sanger-tol/treeval/pull/283

This changes the fasta file path into a tuple containing the meta, fasta and fai.

### Describe alternatives you've considered

_No response_

### Additional context

This will require the module to also be updated to use NF-Test.

The module will not be updated to use the sanger container shown in the linked issue, this is temporary until we can access the main repo to update it properly and includes only an extra flag for buffer_memory manipulation.

### PR DESCRIPTION
**Pull request by DLBPointon**
Adding nf-test tests and files

Adding the option to also include a fai file. If this is not provided then pretext will generate one automatically.
Closes #5123

### CODE REVIEW

File: modules/nf-core/pretextmap/meta.yml [hunk 1]
```diff
@@ -20,6 +20,19 @@ input:
...
+        Groovy Map containing sample information
+        e.g. [ iid:'test2', single_end:false ]
+  - input:
+      type: file
+      description: FASTA file
+      pattern: "*.{fasta,fa}"
+  - input:
+      type: file
+      description: Fasta index
+      pattern: "*fai"
```
Comment (lrauschning) [inline modules/nf-core/pretextmap/meta.yml:35]: Think that should be .fai?

File: modules/nf-core/pretextmap/tests/main.nf.test
Comment (lrauschning) [inline modules/nf-core/pretextmap/tests/main.nf.test:1]: Think testing the behaviour when not passing a `.fai` file would also be good, to ensure there are no future regressions.
Can just be copied from one of the already existing tests while deleting the `.fai` input IMO.

File: modules/nf-core/pretextmap/meta.yml (hunk 1 above)
Comment (DLBPointon) [reply modules/nf-core/pretextmap/meta.yml:35]: You are right, i'll change that now

File: modules/nf-core/pretextmap/tests/main.nf.test
Comment (DLBPointon) [reply modules/nf-core/pretextmap/tests/main.nf.test:1]: No, you are right, I should include a cram + no_fai test at least. Just me being lazy. Testing it now. Thanks for reviewing!

**Comment by yumisims** [comment]
look good to me

File: modules/nf-core/pretextmap/meta.yml [hunk 2]
```diff
@@ -20,6 +20,19 @@ input:
       type: file
       description: BAM/CRAM/SAM file or pairs formatted reads file
       pattern: "*.{bam,cram,sam,pairs.gz}"
+  - input:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ iid:'test2', single_end:false ]
```
Comment (marrip) [inline modules/nf-core/pretextmap/meta.yml:27]: typo?
```suggestion
        e.g. [ id:'test2', single_end:false ]
```

File: modules/nf-core/pretextmap/meta.yml [hunk 3]
```diff
@@ -20,6 +20,19 @@ input:
...
+  - input:
+      type: file
+      description: FASTA file
+      pattern: "*.{fasta,fa}"
```
Comment (marrip) [inline modules/nf-core/pretextmap/meta.yml:31]: I also see `.fna` sometimes

**Review by marrip — APPROVED** [review]
nice work, LGTM! 👍

File: modules/nf-core/pretextmap/meta.yml (hunk 2 above)
Comment (DLBPointon) [reply modules/nf-core/pretextmap/meta.yml:27]: Always at least 1, thanks.

File: modules/nf-core/pretextmap/meta.yml (hunk 3 above)
Comment (DLBPointon) [reply modules/nf-core/pretextmap/meta.yml:31]: Adding now

**Pull request merged**